pkg/process/procutil/data_scrubber_windows.go

```diff
@@ -20,3 +25,61 @@ var (
 	// it's for handling parameters like `/p` and `/rp` in windows
 	forbiddenSymbolsRegex = "[^/a-zA-Z0-9_*]"
 )
+
+var winDotExec = []string{".com", ".exe", ".bat", ".cmd", ".vbs", ".vbe", ".js", ".jse", ".wsf", ".wsh", ".psc1", ".ps1"}
+
+// stripArguments identifies windows extension and extracts command. Otherwise, returns the first element of the cmdline before first space.
+func (ds *DataScrubber) stripArguments(cmdline []string) []string {
+	strCmdline := cmdline[0]
+
+	// Case 1: OS has already completed splitting as there is one token per element, we return first token.
+	if len(cmdline) > 1 {
+		return []string{strCmdline}
+	}
+
+	// Case 2: One string for cmdline, use extensionParser() to find first token.
+	if len(cmdline) == 1 && !strings.HasPrefix(strCmdline, "\"") {
+		strippedCmdline, foundExtension := extensionParser(strCmdline, winDotExec)
+
+		if foundExtension {
+			return []string{strings.TrimSuffix(strippedCmdline, " ")}
+		}
+
+		// If no extension is found, return first token of cmdline.
+		return []string{strings.TrimSuffix(strings.SplitN(strippedCmdline, " ", 2)[0], " ")}
+	}
+
+	// Case 2b: One string for cmdline and first token wrapped in quotes, use findEmbeddedQuotes() to find content between quotes.
+	strippedCmdline := findEmbeddedQuotes(strCmdline)
+	return []string{strings.TrimSuffix(strippedCmdline, " ")}
+}
+
+// extensionParser returns substring of cmdline up to the first extension (inclusive), and boolean on if extension was found.
+// If no extension is found, returns original cmdline.
+// Example: Input="C:\\Program Files\\Datadog\\agent.vbe check process"  Output="C:\\Program Files\\Datadog\\agent.vbe"
+func extensionParser(cmdline string, winDotExec []string) (string, bool) {
```

Comment:
I think it is more idiomatic to either return the executable path, or an empty string. That way you don't need to also return a bool